gui/wxpython/mapdisp/frame.py

```diff
@@ -293,6 +308,25 @@ def CreateStatusbar(self):
                 % dict(command=" ".join(cmd), error=error)
             )
         )
+        return statusbar
+
+    def ShowStatusbar(self, show):
+        """Show/hide statusbar and associated pane"""
+        self.statusbarManager.Show(show)
+        if show:
+            self._mgr.GetPane("statusbar").Show()
+        else:
+            self._mgr.GetPane("statusbar").Hide()
+
+        self._mgr.Update()
+
+    def IsStatusbarShown(self):
+        """Check if statusbar is shown"""
+        return self.statusbarManager.IsShown()
+
```

Comment:
It's enough to show/hide the pane. Also I would remove the unused show functions from sbManager, but check if they are really unused.

```suggestion
    def ShowStatusbar(self, show):
        """Show/hide pane associated with statusbar"""
        self._mgr.GetPane("statusbar").Show(show)
        self._mgr.Update()

    def IsStatusbarShown(self):
        """Check if statusbar is shown"""
        return self._mgr.GetPane("statusbar").IsShown()

```

Reply:
I think it works fine now.